### PR TITLE
Refactor form pattern for change password

### PR DESCRIPTION
**Pull request by jaronheard**
This is still a work in progress, but this demonstrates a working approach that keeps the form logic entirely within the form. It also splits out the fields into separate components, which may or may not be worth it.

There's still some more to do in terms of extracting common components and code, especially for the presentational components.

@andrewbiang888 take a look!

I drew a lot on this article for this pattern: https://techblog.commercetools.com/render-props-apollo-and-formik-build-and-compose-forms-in-react-with-ease-f79a594be239
- It is often best to split concerns into meaningful components before starting to write code.
- Essentially a form should keep the users original input in state until submitted. Moreover, the form should never have to carry out field/input level parsing.
- Validation and conversions should be treated as a form level concern, not on each field or input.

### CODE REVIEW

**Comment by jaronheard** [comment]
@andrewbiang888 - pushed some updates to clean this up even more

**Comment by jaronheard** [comment]
Wrapped up a few of the items that we discussed:
- make Form component take an object of fields
- make common Button, TextField, FormModal presentational components
- make reusable PasswordField component that takes id, fields
- also added propTypes dependency

There's still one potential thing to do for this pattern. I haven't added a separate connector for loading and transforming form data.

Off topic, but I noticed we have both a package-lock.json and yarn.lock? Are we still just using yarn?

File: app/src/components/Forms/Form.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      <Formik
+        initialValues={initialValues}
+        enableReinitialize
+        validationSchema={validationSchema}
+        onSubmit={submitHandler}
+        render={formikProps => {
+          const form = (
+            <React.Fragment>
+              {Object.keys(fields).map(id =>
+                React.createElement(fields[id].component, {
```
Comment (andrewbiang888) [inline app/src/components/Forms/Form.js]: I was wondering how you were going to pull this off. Well done.

**Review by andrewbiang888 — APPROVED** [review]
@jaronheard This looks good. I can't see any conflicts that other input types could cause at this point. I like the application of proptypes and the improve file structure.

**Pull request merged**